apps/studio/src/server/modules/site/site.service.ts

```diff
@@ -303,14 +311,44 @@ export const createSite = async ({ siteName }: CreateSiteProps) => {
         type: "RootPage",
         title: "Home",
       })
+      .onConflict((oc) =>
+        oc.column("draftBlobId").doUpdateSet((eb) => ({
+          draftBlobId: eb.ref("excluded.draftBlobId"),
+        })),
+      )
+      .executeTakeFirstOrThrow()
+  }
 
+  const createSearchPage = async (tx: Transaction<DB>, siteId: number) => {
+    const { id: blobId } = await tx
+      .insertInto("Blob")
+      .values({ content: jsonb(SEARCH_PAGE_BLOB) })
+      .returning("id")
+      .executeTakeFirstOrThrow()
+
+    await tx
+      .insertInto("Resource")
+      .values({
+        draftBlobId: String(blobId),
+        permalink: "search",
+        siteId,
+        type: ResourceType.Page,
+        title: "Search",
+      })
```

Comment:
do we need to set this to `published` so that it'll automatically be picked up on the first build?

Reply:
thought about it and i don't have a preference - i can add it in 👍

Reply:
If we are publishing the search page, can we also publish the homepage as well? So we can have a guarantee that builds will always pass right from the time it is created